File: gateOne/Day seven/Old Task/creditCardValidator.js
const prompt = require("prompt-sync")();
let cardNumber = prompt("Hellow, Kindly Enter Card Number to Verify: ");
	
let firstNumber = cardNumber[0];
let secondNumber = cardNumber[1];
let numberLength = cardNumber.length;
let cardStatus = "";
if(cardNumber.length < 13 || cardNumber.length > 16) 
	cardStatus = ("Invalid Number");
else{
if(firstNumber === ('4'))
	cardStatus = ("Visa Cards");
	
else{
if(firstNumber === ('5'))
	cardStatus = ("Master Card");
	
else{
if(firstNumber === ('6'))
	cardStatus = ("Discover Cards");
	
else{
if(firstNumber === ('3') && secondNumber === ('7'))
	cardStatus = ("American Express Cards");
		
else{
	cardStatus = ("Invalid Number");
	 }
	}
     }
  }
}
	
	let total = 0;
	let totalTwo = 0;
	let cardValidity = "";
	
	for(let count = numberLength - 2;count >= 0;count = count - 2) {
		
		let convertedValue = (cardNumber[count]) - '0';
		if(convertedValue < 0 || convertedValue > 9){
		console.log("Invalid");break;}

		
		let add = convertedValue * 2;
		if(add > 9){
			add = (add / 10) + (add % 10);}
		total += add;
		
	}
	
	for(let count = numberLength - 1;count > 0;count = count - 2) {
		let convertedValue = (cardNumber[count]) - '0';
		
		if(convertedValue < 0 || convertedValue > 9){
		 console.log("Invalid");break;}

		totalTwo += convertedValue;
	}
	if((total + totalTwo) % 10 == 0)
		cardValidity = ("Valid");
	else
		cardValidity =("Invalid");
	
	console.log("\n")
	console.log("****************************************");

	console.log("**Credit Card Type: " + cardStatus);

	console.log("**Credit Card Number: " + cardNumber);

	console.log("**Credit Card Digit: " + numberLength);

	console.log("**Credit Card Validity Status: " + cardValidity);

	console.log("****************************************");
	

File: gateOne/Day seven/Old Task/credit_card_validator.py
cardNumber = input("Hellow, Kindly Enter Card Details to Verify\n")
	
firstNumber = cardNumber[0]
secondNumber = cardNumber[1]
numberLength = len(cardNumber)
cardStatus = ""
if(len(cardNumber) < 13 or len(cardNumber) > 16):
	cardStatus = ("Invalid Card")
elif(firstNumber == ('4')):
	cardStatus = ("Visa Cards")
	
elif(firstNumber == ('5')):
	cardStatus = ("Master Card")
	
elif(firstNumber == ('6')):
	cardStatus = ("Discover Cards")
	
elif(firstNumber == ('3') and secondNumber == ('7')):
	cardStatus = ("American Express Cards")
		
else:
	cardStatus = ("Invalid Card")
	
total = 0
totalTwo = 0
cardValidity = ""
for count in range((numberLength - 2),-1,-2):
		
	convertedValue = int(cardNumber[count])
	if(convertedValue < 0 or convertedValue > 9):
		print("Invalid")
		break
	
	add = convertedValue * 2
	if(add > 9):
		add = (add // 10) + (add % 10)
	total += add
	
for count in range((numberLength - 1),0,-2):
	convertedValue = int(cardNumber[count])
	if(convertedValue < 0 or convertedValue > 9):
		print("Invalid")
		break

	totalTwo += convertedValue
	
if((total + totalTwo) % 10 == 0):
	cardValidity = ("Valid")
else:
	cardValidity =("Invalid")
		
print("\n")
print("****************************************")

print("**Credit Card Type: " + cardStatus)

print("**Credit Card Number: " + cardNumber)

print(f"**Credit Card Digit:  {numberLength}")

print("**Credit Card Validity Status: " + cardValidity)

print("****************************************")
	
	
	
	
